src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -252,13 +304,29 @@ class NBDServer
 	  dout(0) << "disconnect request received" << dendl;
           return;
         case NBD_CMD_WRITE:
+#ifdef CEPH_HAVE_SPLICE
+          // 1<<17 is not a magic number. It is value that causes splice not to wait if there is less data.
+          // See test/bufferlist, splice_Nagle_conflict. This value is derived directly from there.
+          // When future linux releases got rid of this problem, we should change/get rid of it.
+          if (ctx->request.len >= (1<<17)) {
```

Comment:
@dillaman Got rid of magic value. Now all complexity is confined to insert_from_fd operation on bufferlist.